### PR TITLE
Private/eszkadev/annotation fixes

### PR DESCRIPTION
**Pull request by eszkadev**
annotations: isAutoSaved helper

- we need to check if received comment is "our" autosaved one
- without that we were showing comments from others on the
  screen when we were in "autosaved mode"
- remove "interval" to make it more deterministic

This fixes regression from commit 93bdcc349b75ecc0a5268f2577504e8e5ba3dfee
Annotation: revamp comment autosaving

### CODE REVIEW

File: browser/src/layer/tile/CommentSection.ts [hunk 1]
```diff
@@ -754,6 +756,18 @@ export class Comment extends CanvasSectionObject {
...
 
+	// check if this is "our" autosaved comment
+	// core is not aware it's autosaved one so use this simplified detection based on content
+	public isAutoSaved (): boolean {
+		var autoSavedComment = cool.CommentSection.autoSavedComment;
+		if (!autoSavedComment)
+			return false;
+
+		var authorMatch = autoSavedComment.sectionProperties.data.author === this.sectionProperties.data.author;
+		var textMatch = autoSavedComment.sectionProperties.data.text === this.sectionProperties.data.text;
```
Comment (lpranam) [inline browser/src/layer/tile/CommentSection.ts:767]: This makes reply autosave fail in writer. 

When we reply to any comment we edit into the original comment element in reply textbox, so original comment is set as `autoSavedComment` value and `this.sectionProperties.data.text` is new reply, so these values are not going to match for reply.

**Review by lpranam — CHANGES_REQUESTED** [review]

File: browser/src/layer/tile/CommentSection.ts (hunk 1 above)
Comment (eszkadev) [reply browser/src/layer/tile/CommentSection.ts]: ok, then we should match only author

**Comment by eszkadev** [comment]
fails in `impress/annotation_spec.js`

**Comment by lpranam** [comment]
fixed failing test

**Review by lpranam — APPROVED** [review]
Looks good to me

**Pull request merged**